internal/pkg/gateway/commit/channelnotifier.go

```diff
@@ -12,85 +12,113 @@ import (
 	"github.com/hyperledger/fabric/core/ledger"
 )
 
-type channelNotifier struct {
+type channelLevelNotifier struct {
 	lock          sync.Mutex
 	commitChannel <-chan *ledger.CommitNotification
-	listeners     map[*channelListener]bool
+	listeners     map[string][]*channelLevelListener
 }
 
-func newChannelNotifier(commitChannel <-chan *ledger.CommitNotification) *channelNotifier {
-	notifier := &channelNotifier{
+func newChannelNotifier(commitChannel <-chan *ledger.CommitNotification) *channelLevelNotifier {
+	notifier := &channelLevelNotifier{
 		commitChannel: commitChannel,
-		listeners:     make(map[*channelListener]bool),
+		listeners:     make(map[string][]*channelLevelListener),
 	}
 	go notifier.run()
 	return notifier
 }
 
-func (channelNotifier *channelNotifier) run() {
+func (notifier *channelLevelNotifier) run() {
```

Comment:
better, but still atypically long for a self reference

Reply:
I personally am not a fan of single character variables names other than specific, very well established (and generally cross-language) conventions such as i as a loop index.

If you are mandating that code must use single-character receiver names to be allowed into Fabric, please can you come out and say that so I can understand clearly, and create linting rules to require this.

Reply:
That's fine. Where the argument starts to break apart is when there are different patterns within the same component. The gateway has:

- func (ef *endpointFactory)
- func (gs *Server)
- func (e *EndorserServerAdapter)

Seems there should at least be consistency within this gateway effort. Don't you agree?

Reply:
I can read the code ok either way so no, I’m not that worried about consistency here. If you are then my personal preference would be to avoid single-character variable names throughout. I am happy to make that change in this PR or a separate one

Reply:
Honestly, I'd rather you adopt the conventions of the project instead of your own preferences. There's a reason why there's so much inconsistency in our configuration, our structure, and our tests. We have started to settle on some common preferences and I'd very much like to see green field code adhere to them.

Reply:
You're right, these (any many other PR comments) are personal preference. I can absolutely relate; it would be nice for me if everyone had exactly the same personal preferences as me when it comes to code. But they don't, and for me that's fine provided the code is functionally correct (as demonstrated by passing tests) and fairly sane (along the lines of Kent Beck's rules of simple design). I try not to impose my personal preferences on other people's code. Nobody wants to contribute to a project -- particularly an open source project like this -- if they are allowed no freedom to make their own choices about how they implement their code